File: pa5/DictionaryTest.c
// -------------------------------------------
// Duy Nguyen
// duminguy
// CS 12B pa5
// DictionaryTest.c
// Test client for Dictionary class
// -------------------------------------------

#include <stdio.h>
#include <stdlib.h>
#include "Dictionary.h"

#define MAX_LEN 180

int main(void){
    Dictionary A = newDictionary();
    
    insert(A, 1, 10);
    insert(A, 2, 20);
    insert(A, 3, 30);
    insert(A, 4, 40);
    insert(A, 5, 50);
    
    printDictionary(stdout, A);
    printf("%s\n", (isEmpty(A)?"true":"false"));
    printf("%d\n", size(A));
    char* v = lookup(A, 3);
    printf("key=\"%s\n", (v==NULL?"not found ":"value="), v);
    
    delete(A, 2);
    delete(A, 4);
    printf("%s\n", (isEmpty(A)?"true":"false"));
    printf("%d\n", size(A));
    char* v = lookup(A, 3);
    printf("key=\"%s\n", (v==NULL?"not found ":"value="), v);
    
    makeEmpty(A);
    printf("%s\n", (isEmpty(A)?"true":"false"));
    printf("%d\n", size(A));
    char* v = lookup(A, 3);
    printf("key=\"%s\n", (v==NULL?"not found ":"value="), v);

    freeDictionary(&A);
    
}

File: pa3/DictionaryTest.java
// --------------------------------------
// Duy Nguyen
// W1475318
// CS 12B pa3
// DictionaryTest.java
// Test client for Dictionary class
// --------------------------------------

public class DictionaryTest{

    public static void main(String[] args){
	Dictionary A = new Dictionary();
	
	System.out.println(A.isEmpty());
	System.out.println("The size is: " + A.size());
	A.delete("1");
	A.makeEmpty();
	System.out.println(A.toString());

	A.insert("1", "a");
	System.out.println(A.isEmpty());
	System.out.println("The size is: " + A.size());
	System.out.println(A.lookup("2"));
	System.out.println(A.lookup("1"));
	System.out.println(A.toString());
	A.delete("3");
	A.delete("1");
	A.makeEmpty();
	System.out.println(A.isEmpty());
	System.out.println("The size is: " + A.size());
	
	A.insert("1", "a");
	A.insert("2", "b");
	A.insert("3", "c");
	A.insert("4", "d");
	A.insert("5", "e");
	A.insert("5", "f");
	System.out.println(A.isEmpty());
	System.out.println("The size is: " + A.size());
	System.out.println(A.lookup("3"));
	System.out.println(A.toString());
	A.delete("3");
	System.out.println("The size is: " + A.size());
	System.out.println(A.lookup("3"));
	System.out.println(A.isEmpty());
	A.makeEmpty();
	System.out.println("The size is: " + A.size());
	System.out.println(A.isEmpty());
	
    }

}
